### DISCUSSION TITLE
Naming and capitalization

### DISCUSSION THREAD

**Discussion by RalfG** [IanTerzo/Neutron#15] [General]
Hi @IanTerzo,

Great project! It really lowers the bar for implementing a pywebview GUI.

I was wondering if there is still some flexibility regarding the naming of the package and functions/methods in the Python API? Mostly because not everything seems to follow [Python conventions and PEP8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names):

- **Package name**
  I understand that the `neutron` package name (as listed on PyPI) is already taken, so `neutron-web` makes a lot of sense as alternative. However, I would propose to keep the name in lowercase, following PEP8.

- **Module name**
  For consistency with the package name, and to avoid confusion with the OpenStack module `neutron`, I would propose `neutron_web`; although `neutron` would also work (potential clashes can be avoided with `import neutron as neutron_web`). In any case, I definitely advocate to use lowercase and underscores, as PEP8 proposes.

- **Methods/functions**
  Same here as with the module name, PEP8 proposes fully lower case and underscores. For consistency with other Python projects, I would change method and function names, such as `HTMLelement`'s `AddEventListener()` to `add_event_listener()`. This helps users to differentiate classes (uppercase) with functions and methods (lowercase with underscores).

Looking forward to hear your thoughts on this.

Best,
Ralf

**Comment by IanTerzo** [comment]
Hello, thanks for reaching out.

I think that keeping the package name in lowercase is great idea and will change to it soon, however I am not sure about changing the module name to `neutron_web`. I if keep the current name, `Neutron`, I will make it underscore to follow PEP8 as you stated. Regarding the methods/functions they are named like that to be identical to those in the HTML DOM, (the capital "A" in  `AddEventListener `is a typo BTW).
